### PR TITLE
update logic for blocking users from Direct Deposit

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
We had apparently been looking at the wrong flag to determine if we should block Vets from the Direct Deposit feature on the Profile. It's likely we have been showing a lot of GIB recipients that they were not allowed access to the Direct Deposit feature :-/

## Testing done
Local + updated unit tests

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/profile360/selectors.js [hunk 1]
```diff
@@ -29,8 +29,15 @@ export const directDepositAddressIsSetUp = state => {
...
-  if (!controlInfo) return false;
-  return controlInfo.canUpdateAddress !== true;
+  const controlInfo =
+    directDepositInformation(state)?.responses?.[0]?.controlInformation || {};
+  // If all of these flags are not strictly equal to `true` we want to block
+  // access to the direct deposit feature. ie, we want to err on the side of
+  // blocking people from accessing direct deposit rather than err on the side
+  // of letting people have access to the feature.
+  return (
+    controlInfo.isCompetentIndicator !== true ||
```
Comment (sanlouise) [inline src/applications/personalization/profile360/selectors.js:39]: Can we check here for `false` values instead? 😄

File: src/applications/personalization/profile360/tests/selectors.unit.spec.js [hunk 2]
```diff
@@ -118,36 +118,76 @@ describe('profile360 selectors', () => {
...
                 },
               },
             ],
           },
         },
       };
-      expect(selectors.directDepositIsBlocked(state)).to.be.false;
+      expect(selectors.directDepositIsBlocked(state)).to.be.true;
+    });
+    it('returns `true` if the `noFiduciaryAssignedIndicator` is not true', () => {
```
Comment (sanlouise) [inline src/applications/personalization/profile360/tests/selectors.unit.spec.js:166]: Can we swap all references of `not true` to `false`? 😃

File: src/applications/personalization/profile360/selectors.js (hunk 1 above)
Comment (sanlouise) [reply src/applications/personalization/profile360/selectors.js:39]: Looking at your unit tests, is it either `true` or `null`?

Comment (sanlouise) [reply src/applications/personalization/profile360/selectors.js:39]: Can we instead write this? It's just a bit hard to follow otherwise

`!controlInfo.isCompetentIndicator || !controlInfo.noFiduciaryAssignedIndicator || !controlInfo.notDeceasedIndicator`

Comment (erikphansen) [reply src/applications/personalization/profile360/selectors.js]: this code actually came from a previous commit and at the time I, for some reason, wanted to make sure each value was strictly true rather than checking for truthy or not falsey. I don't remember why. But the tests still pass with the simpler check you suggested and I'm happy with this tests. So 👍

Comment (sanlouise) [reply src/applications/personalization/profile360/selectors.js]: Awesome! 👍

**Review by sanlouise — APPROVED** [review]
